packages/xstate-graph/src/graph.ts

```diff
@@ -675,3 +604,20 @@ export function traverseShortestPathsFromTo<TState, TEvent extends EventObject>(
     );
   });
 }
+
+export function joinPaths<TState, TEvent extends EventObject>(
+  path1: StatePath<TState, TEvent>,
+  path2: StatePath<TState, TEvent>
+): StatePath<TState, TEvent> {
+  const secondPathSource = path2.steps[0]?.state ?? path2.state;
+
+  if (secondPathSource !== path1.state) {
+    throw new Error(`Paths cannot be joined`);
+  }
+
+  return {
+    state: path1.state,
```

Comment:
Shouldn't this hold the "target"/destination state? If yes then we should use the `path2.state` as the final joined path is going to end up in that.
```suggestion
    state: path2.state,
```

Reply:
Fixed + test added in [963b429](https://github.com/statelyai/xstate/pull/3446/commits/963b429d91a9dbfe3ac9f0a95a4ccef911a79cdf)